mage_ai/data_cleaner/column_types/column_type_detector.py

```diff
@@ -38,34 +42,46 @@ def find_syntax_errors(series, column_type):
     if len(series) == 0:
         return pd.Series([])
     dtype = series.dtype
-    str_series = series
+    invalid_placeholder = INVALID_VALUE_PLACEHOLDERS[column_type]
+    null_placeholder = CONSTANT_IMPUTATION_DEFAULTS[column_type]
+    mask = pd.Series([False] * len(series))
+    mask.index = series.index
     if column_type == ColumnType.EMAIL:
-        pattern = REGEX_EMAIL
+        placeholder_matches = series.str.match(invalid_placeholder, na=True)
```

Comment:
Could you reduce duplicate code in this method?

Reply:
Refactored to reduce duplicated code + added a new test case to make sure that the added logic works correctly.